### DISCUSSION TITLE
How do I insert a custom A tag like <a href="url" name="test" linkId="testId"></a>  tag

### DISCUSSION THREAD

**Discussion by lzxlzxlzxl** [ueberdosis/tiptap#3950] [General Questions]
How do I insert a custom A tag like <a href="url" name="test" linkId="testId"></a>  tag

**Comment by codemzy** [comment]
You could extend the current link extension to add the custom `name` and `linkId` attributes...

```js
import Link from '@tiptap/extension-link';

// add custom attributes to link
export const CustomLink = Link.extend({
    addAttributes() {
        return {
            ...this.parent?.(),
            name: {
                default: null,
            },
            linkId: {
                default: null,
            },
        }
    },

    renderHTML({ HTMLAttributes }) {
        return ['a', HTMLAttributes];
    },
});
```

**Comment by EminAbdullayev1998** [comment]
You can create a custom <a> tag with the href, name, and linkId attributes:
`<a href="url" name="test" data-link-id="testId">Link Text</a>`

**In this example, `href` specifies the URL that the link should point to, `name` specifies the name of the link (which can be used as an anchor link), and `data-link-id` is a custom attribute that you can use to store additional information about the link.**
